src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/RenameStrategies/AbstractRenameStrategy.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using System.Globalization;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.RenameStrategies
+{
+    internal abstract class AbstractRenameStrategy : IRenameStrategy
+    {
+        protected readonly IProjectThreadingService _threadingService;
+        protected readonly IUserNotificationServices _userNotificationServices;
+        protected readonly IOptionsSettings _optionsSettings;
+        private bool _userPromptedOnce = false;
+        private bool _userConfirmedRename = true;
+
+        public AbstractRenameStrategy(IProjectThreadingService threadingService, IUserNotificationServices userNotificationService, IOptionsSettings optionsSettings)
+        {
+            _threadingService = threadingService;
+            _userNotificationServices = userNotificationService;
+            _optionsSettings = optionsSettings;
+        }
+
+        public abstract bool CanHandleRename(string oldFilePath, string newFilePath);
+
+        public abstract Task RenameAsync(Project newProject, string oldFilePath, string newFilePath);
+
+        protected async Task<bool> CheckUserConfirmation(string oldFileName)
+        {
+            if (_userPromptedOnce)
```

Comment:
This is related to #272 , When we have more than one matching symbols , we want to change all but prompt user only once.